src/events.js

```diff
@@ -44,19 +44,28 @@ Object.keys(eventMap).forEach(key => {
     Object.assign(node, targetProperties)
     const window = getWindowFromNode(node)
     const EventConstructor = window[EventType] || window.Event
+    let event
     /* istanbul ignore else  */
     if (typeof EventConstructor === 'function') {
-      return new EventConstructor(eventName, eventInit)
+      event = new EventConstructor(eventName, eventInit)
     } else {
       // IE11 polyfill from https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent#Polyfill
-      const event = window.document.createEvent(EventType)
+      event = window.document.createEvent(EventType)
       const {bubbles, cancelable, detail, ...otherInit} = eventInit
       event.initEvent(eventName, bubbles, cancelable, detail)
       Object.keys(otherInit).forEach(eventKey => {
         event[eventKey] = otherInit[eventKey]
       })
-      return event
     }
+
+    // Approximate dataTransfer on the event object
+    // jsdom does not support DataTransfer constructor
```

Comment:
We don't only support JSDOM, so let's try the DataTransfer constructor first and if that's unsupported *then* we can do this workaround. Sound ok?

That being the case, we'll probably need to add `/* istanbul ignore if */` or `/* istanbul ignore else */` to maintain 100% code coverage.